k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java

```diff
@@ -0,0 +1,140 @@
+package com.fsck.k9.activity;
+
+import com.fsck.k9.Account;
+import com.fsck.k9.mail.Message;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static org.junit.Assert.assertNotEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = "src/main/AndroidManifest.xml", sdk = 21)
+public class ActivityListenerTest {
+
+    private ActivityListener activityListener;
+    private Account account;
+    private String folder;
+    private String errorMessage;
+    private Message message;
+    private int count;
+
+    @Before
+    public void before() {
+        account = mock(Account.class);
+        when(account.getDescription()).thenReturn("account");
+        folder = "folder";
+        errorMessage = "errorMessage";
+        message = mock(Message.class);
+        count = 1;
+        activityListener = new ActivityListener();
+    }
+
+    @Test
+    public void folderStatusChanged__shouldResultInValidStatus() {
```

Comment:
You're testing `getOperation()`. So `getOperation_afterFolderStatusChanged_shouldResultInValidStatus` is probably a better name. Personally, I'd just drop the last part as it basically says "should work as expected", which is implied when writing tests.

Also, I don't like the double underscore. It's bad enough we have to use underscores for readability. Having two of them is unnecessary, though. /cc @Valodim